spec/src/main/asciidoc/query-language.asciidoc

```diff
@@ -485,10 +485,42 @@ A `select` statement, with syntax given by `select_statement`, returns data to t
 
 An `update` statement, with syntax given by `update_statement`, updates each record which satisfies the restriction imposed by the `where` clause, and returns the number of updated records to the client.
 
+[WARNING]
+====
+Support for update statements may vary across NoSQL databases.
+
+In particular:
+
+- **Key-Value databases** typically only support updates by key. They do not allow conditional updates using a `where` clause, and returning the number of updated records is generally unsupported.
+- **Wide-Column databases** may restrict updates to records identified by primary or partition keys. Conditional updates or returning update counts may not be supported unless secondary indexes are used.
+- **Document and Graph databases** are more likely to support conditional updates and return the number of updated records. However, the semantics and consistency guarantees may vary depending on the provider.
+
+Due to the **eventual persistence** behavior common in many NoSQL databases, returning the number of updated records as an `int` or `long` might not be supported. Even when supported, the value may be **imprecise**, reflecting an estimate or partial result rather than an exact count.
```

Comment:
```suggestion
Due to the _eventual persistence_ behavior common in many NoSQL databases, returning the number of updated records as an `int` or `long` might not be supported. Even when supported, the value may be imprecise, reflecting an estimate or partial result rather than an exact count.
```